blocks/procedures.ts

```diff
@@ -528,24 +546,29 @@ blocks['procedures_defreturn'] = {
   },
   /**
    * Return the signature of this procedure definition.
-   * @return {!Array} Tuple containing three elements:
+   *
+   * @returns Tuple containing three elements:
    *     - the name of the defined procedure,
    *     - a list of all its arguments,
    *     - that it DOES have a return value.
    * @this {Block}
    */
-  getProcedureDef: function() {
+  getProcedureDef: function(this: ProcedureBlock): [string, string[], true] {
     return [this.getFieldValue('NAME'), this.arguments_, true];
   },
   callType_: 'procedures_callreturn',
 };
 
-blocks['procedures_mutatorcontainer'] = {
+/** Type of a procedures_mutatorcontainer block. */
+type ContainerBlock = Block&ContainerMixin;
+interface ContainerMixin extends ContainerMixinType {}
+type ContainerMixinType = typeof PROCEDURES_MUTATORCONTAINER;
+
+const PROCEDURES_MUTATORCONTAINER = {
   /**
    * Mutator block for procedure container.
-   * @this {Block}
    */
-  init: function() {
+  init: function(this: Block) {
```

Comment:
`this: ContainerBlock` ?

Reply:
Done.  Obviously not necessary, but does help make things clearer.  (I think I had to go back and add `ContainerBlock` for the benefit of later block types, and didn't update this one.)